modules/structured_light/samples/sl_example.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/structured_light.hpp>
+
+using namespace std;
+using namespace cv;
+
+// 1. Calibrate camera=projector
+// 2. Save calibration params
+// 3. Generate projected pattern
+// 4. Distort projected pattern
+// 5. Capture patterns reference pattern - optional
+// 6. Capture patterns with object
+// 7. Image preprocessing
+// 7. Load refs and images to unwrap algorithm
+// 8. Save points to file
+
+void readImages(vector<string> refs_files, vector<string> imgs_files, OutputArrayOfArrays refs, OutputArrayOfArrays imgs)
+{
+    vector<Mat>& refs_ = *(vector<Mat>*) refs.getObj();
+    vector<Mat>& imgs_ = *(vector<Mat>*) imgs.getObj();
```

Comment:
> `.getObj()`

This call must be avoided. It is not safe (especially without checks).

It is better to replace `OutputArrayOfArrays` to `vector<Mat>&` in arguments list (it is fine for sample code, we are not needed universal wrapper here).